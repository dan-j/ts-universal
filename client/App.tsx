import React from 'react';
import * as history from 'history';
import universal from 'react-universal-component';

const styles = require('./App.css');

interface AppProps {
    name?: String;
    history: history.History;
}


const Home = universal(() => import('./Home'), {
    minDelay: 1200,
    loading: <div>loading</div>,
    error: <div>Error</div>,
});

const Admin = universal(() => import('./Admin'), {
    minDelay: 1200,
    loading: <div>loading</div>,
    error: <div>Error</div>,
});

export default class App extends React.Component<AppProps, {}> {

    static defaultProps = {
        name: 'World',
    };

    goHome = () => {
        this.props.history.push('/home');
        this.setState({});
    };

    goAdmin = () => {
        this.props.history.push('/admin');
        this.setState({});
    };

    render() {
        const { name, history } = this.props;
        const { location } = history;
        return (
            <div>
                <p className={styles.greeting}>Hello, {name}</p>
                {location.pathname === '/home' && <Home />}
                {location.pathname === '/admin' && <Admin />}
                <button onClick={this.goHome}>
                    home
                </button>
                <button onClick={this.goAdmin}>
                    admin
                </button>
            </div>
        );
    }
}
