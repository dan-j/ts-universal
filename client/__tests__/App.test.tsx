import React from 'react';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history';
import toJson from 'enzyme-to-json';

import App from '../App';

describe('<App />', () => {
    it('Renders a default greeting', () => {
        const wrapper = shallow(<App history={createMemoryHistory()} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Renders a personalised greeting', () => {
        const wrapper = shallow(<App name="Dan" history={createMemoryHistory()} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});

