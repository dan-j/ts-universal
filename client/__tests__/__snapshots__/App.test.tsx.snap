// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders a default greeting 1`] = `
<div>
  <p
    className="greeting"
  >
    Hello, 
    World
  </p>
  <button
    onClick={[Function]}
  >
    home
  </button>
  <button
    onClick={[Function]}
  >
    admin
  </button>
</div>
`;

exports[`<App /> Renders a personalised greeting 1`] = `
<div>
  <p
    className="greeting"
  >
    Hello, 
    Dan
  </p>
  <button
    onClick={[Function]}
  >
    home
  </button>
  <button
    onClick={[Function]}
  >
    admin
  </button>
</div>
`;
